Source/Urho3D/Resource/GraphPin.cpp

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright (c) 2021 the Urho3D project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "GraphPin.h"
+
+#include "Graph.h"
+#include "GraphNode.h"
+
+#include "GraphNode.h"
+#include "../IO/ArchiveSerialization.h"
+#include "../IO/Log.h"
+
+namespace Urho3D
+{
+
+GraphPin::GraphPin(GraphNode* node, GraphPinDirection direction)
+    : node_(node)
+    , direction_(direction)
+    , type_(VAR_NONE)
+{
+}
+
+void GraphPin::SetType(VariantType type)
+{
+    type_ = type;
+}
+
+bool GraphPin::Serialize(Archive& archive)
```

Comment:
General comment about Serialize signatures: `Serialize(Archive& archive)` is expected to write exactly one value or block.
Use signature `Serialize(Archive& archive, ArchiveBlock& block)` if you intend to write multiple values.

I.e. `Serialize(Archive& archive)` is "write an object as whole" and should work unconditionally at any moment
and `Serialize(Archive& archive, ArchiveBlock& block)` is "write data into prepared block" and it may make assumptions about destination.